File: 412/fizz-buzz.js
/**
 * @param {number} n
 * @return {string[]}
 */
var fizzBuzz = function (n) {
    let res = [];
    if (n == 0) {
        return res;
    }

    for (let i = 0; i < n; i++) {
        let [fizzQ, buzzQ] = [(i + 1) % 3, (i + 1) % 5];

        if (!fizzQ && !buzzQ) {
            res.push("FizzBuzz");
        } else if (!fizzQ) {
            res.push("Fizz");
        } else if (!buzzQ) {
            res.push("Buzz");
        } else {
            res.push(`${i + 1}`);
        }
    }
    return res;
};

File: 412/fizz-buzz.py
class Solution:
    def fizzBuzz(self, n):
        """
        :type n: int
        :rtype: List[str]
        """
        res = []
        for i in range(n):
            if (i+1) % 15 == 0:
                res.append("FizzBuzz")
            elif (i+1) % 3 == 0:
                res.append("Fizz")
            elif (i+1) % 5 == 0:
                res.append("Buzz")
            else:
                res.append(str(i+1))

        return res
